From e8775cc579a2104ddb4d6c11a7fef7732261d205 Mon Sep 17 00:00:00 2001
From: Damien Mascord <tusker@tusker.org>
Date: Mon, 1 Nov 2021 20:06:20 +1100
Subject: [PATCH] build: fix ldconfig executable error in python

The empty executable is causing problems with meson builds, due to the
error: OSError: [Errno 8] Exec format error: 'ldconfig'

This patch changes the empty ldconfig stub to symlink to /bin/true to
work around this issue.

Fixes: FS#4117

Signed-off-by: Damien Mascord <tusker@tusker.org>
---
 include/prereq-build.mk | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/include/prereq-build.mk b/include/prereq-build.mk
index d28432f726..d3593b0205 100644
--- a/include/prereq-build.mk
+++ b/include/prereq-build.mk
@@ -196,5 +196,4 @@ prereq: $(STAGING_DIR_HOST)/bin/mkhash
 
 # Install ldconfig stub
 $(eval $(call TestHostCommand,ldconfig-stub,Failed to install stub, \
-	touch $(STAGING_DIR_HOST)/bin/ldconfig && \
-	chmod +x $(STAGING_DIR_HOST)/bin/ldconfig))
+	$(LN) /bin/true $(STAGING_DIR_HOST)/bin/ldconfig))
-- 
2.35.1

